modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
+        if !collapsedDim(d) {
+          downDim += 1;
+          if (upDim == downDim) then
+            return downdom.dsiDim(d);
+        }
+      }
+      halt("Fell out of loop in dsiDim():", (upDim, downDim));
+    }
+
+    proc dsiDims() {
+      var upDims: rank*range(idxType, stridable=stridable);
+      var upDim = 1;
+      for param downDim in 1..downrank {
+        if !collapsedDim(downDim) {
+          upDims(upDim) = downdom.dsiDim(downDim);
+          upDim += 1;
+        }
+      }
+      return upDims;
+    }
+
+    proc dsiGetIndices() {
+      return updom.dsiGetIndices();
+    }
+
+    proc dsiSetIndices(inds) {
+      pragma "no auto destroy"
+      var updomRec = {(...inds)};
+      updom = updomRec._value;
+      var downdomclass = dist.downdist.newRectangularDom(rank=downrank,
+                                                         idxType=idxType,
+                                                         stridable=stridable);
+      pragma "no auto destroy"
+      var downdomLoc = _newDomain(downdomclass);
+      downdomLoc = chpl_rankChangeConvertDom(inds, inds.size, collapsedDim, idx);
+      downdomLoc._value._free_when_no_arrs = true;
+      downdomPid = downdomLoc._pid;
+      downdomInst = downdomLoc._instance;
+    }
+
+    proc dsiMember(i) {
+      return downdom.dsiMember(chpl_rankChangeConvertIdx(i, collapsedDim, idx));
+    }
+
+    iter these() {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom do
+          yield i;
+      } else {
+        for i in downdom do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.standalone {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag) do
+          yield i;
+      } else {
+        for i in downdom.these(tag) do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.leader {
+      if chpl__isDROrDRView(downdom) {
+        for followThis in updom.these(tag) do
+          yield followThis;
+      } else {
+        for followThis in downdom.these(tag) {
+          const followThisLoD = chpl_rankChangeConvertHiDTupleToLoD(followThis);
+          yield followThisLoD;
+        }
+      }
+    }
+
+    iter these(param tag: iterKind, followThis)
+      where tag == iterKind.follower {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag, followThis) do
+          yield i;
+      } else {
+        const followThisHiD = chpl_rankChangeConvertLoDTupleToHiD(followThis);
+        for i in downdom.these(tag, followThisHiD) {
+          yield chpl_rankChangeConvertIdxHiDToLoD(i, collapsedDim, idx, rank);
+        }
+      }
+    }
+
+    proc chpl_rankChangeConvertLoDTupleToHiD(tup) {
+      var tupHiD: downrank*tup(1).type;
+      var i = 1;
+      for param d in 1..downrank do
+        if collapsedDim(d) then
+          tupHiD(d) = 0..0;  // TODO: This seems bogus/arbitrary... or is it?
```

Comment:
What I'm finding is:
* if idxType is int, the follower works as expected
* if idxType is not int, lots of other things fall apart before ever getting here

For now, I'm going to add a TODO to try and fix bugs related to idxType != int and remove the comment since I'm more confident that it's correct.  Keep in mind that this is in the context of a leader/follower iterator's 'followThis' argument and that we 0-shift and densify the index sets in such cases so that arrays/domains of the same shape/size but different index sets can be zipped.  So for a singleton dimension, I believe that 0..0 is correct.  Though I don't recall offhand, I'd guess that we also have the 'followThis' argument be represented using 'int' ranges regardless of idxType so that arrays/domains of different index types can be zipped together (and arguably we could/should use 'uint' since we're not using negative numbers in this context and _may_ want the full range of uints available to us.  But that's independent of this PR.